lib/ClangImporter/ClangImporter.cpp

```diff
@@ -7471,6 +7471,75 @@ bool importer::isForeignReferenceTypeWithoutImmortalAttrs(const clang::QualType
          !hasImmortalAtts(pointeeType->getDecl());
 }
 
+llvm::SmallVector<const clang::RecordDecl *, 4>
+getMatchingParentDecls(const clang::RecordDecl *decl) {
+  assert(decl && "decl is null inside getMatchingParentDecls");
+
+  llvm::SmallVector<const clang::RecordDecl *, 4> matchingDecls;
+
+  // Base case
+  if (hasImportAsRefAttr(decl)) {
+    matchingDecls.push_back(decl);
+  }
+
+  if (const auto *cxxRecordDecl = llvm::dyn_cast<clang::CXXRecordDecl>(decl)) {
+    if (!cxxRecordDecl->hasDefinition())
+      return matchingDecls;
+    cxxRecordDecl->forallBases([&](const clang::CXXRecordDecl *baseDecl) {
+      if (baseDecl->hasDefinition() && hasImportAsRefAttr(baseDecl)) {
```

Comment:
Why do we need to have a definition for this attribute to work?

Reply:
I was getting some assertion failures in the lit tests if I was directly calling `hasImportAsRefAttr` on `decl`s which do not have a definition. So I started checking that `decl`s have definition before they are processed. Can there be scenarios where this check could be problematic? @Xazax-hun  Or should we diagnose differently for this case?

The first check(`!cxxRecordDecl->hasDefinition()`)  is absolutely necessary. I can probably remove `baseDecl->hasDefinition()`

Reply:
I think it is fine to have this check. We probably do not support importing incomplete types anyway. I think this is a feature that we could potentially have for foreign reference types, but let's not do anything in this direction unless we have a specific as from our users.